apps/coordinator/ts/scheduler/dto.ts

```diff
@@ -0,0 +1,95 @@
+import { EMode } from "@maci-protocol/sdk";
+import { ApiProperty } from "@nestjs/swagger";
+import { IsEnum, IsEthereumAddress, IsInt, IsOptional, IsString, Min } from "class-validator";
+
+import type { Hex } from "viem";
+
+import { ESupportedNetworks } from "../common";
+
+/**
+ * Data transfer object for scheduled poll
+ */
+export class IdentityScheduledPollDto {
+  /**
+   * MACI contract address
+   */
+  @ApiProperty({
+    description: "MACI contract address",
+    type: String,
+  })
+  @IsEthereumAddress()
+  maciAddress!: Hex;
+
+  /**
+   * Poll id
+   */
+  @ApiProperty({
+    description: "Poll id",
+    minimum: 0,
+    type: Number,
+  })
+  @IsInt()
+  @Min(0)
+  pollId!: number;
+
+  /**
+   * Chain Name
+   */
+  @ApiProperty({
+    description: "Chain to which to deploy the contract(s)",
+    enum: ESupportedNetworks,
+  })
+  @IsEnum(ESupportedNetworks)
+  chain!: ESupportedNetworks;
+}
+
+/**
+ * Data transfer object for registering a poll for finalization
+ */
+export class SchedulePollWithSignerDto extends IdentityScheduledPollDto {
```

Comment:
Why is it called `SchedulePollWithSignerDto` and `sessionKeyAddress` and `approval` are optional?

Reply:
`sessionKeyAddress` and `approval` are used for the AA signer (that uses session keys). The AA signer was giving us some problems in the deploy poll step so we decided to use a normal ethers js signer if `sessionKeyAddress` and `approval` were not provided.

I hope we fix the issues with the AA signers and users will be able to use the normal ethers js and the AA signer. I have an issue to keep track of this: https://github.com/privacy-scaling-explorations/maci/issues/2638